### PR TITLE
Add `rootstock` chain

### PR DESCRIPTION
**Pull request by ahsan-javaiid**
### Description
Adds Support for [rootstock](https://rootstock.io/) chain.

### How to test
Same way as other EVM chains

### Types of changes
Checklist
- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by ahsan-javaiid** [comment]
cc: @alepc253

**Comment by Milerius** [comment]
Hey, please add your network using custom network feature on mobile.

**Comment by ahsan-javaiid** [comment]
Hi @Milerius

I created this pull request while doing integration with [alpha-wallet](https://github.com/AlphaWallet/alpha-wallet-android) which uses this wallet-core sdk. And `rootstock` is not supported by wallet-core due to which integration will be incomplete. 

I added rootstock as custom network but still we need this integration to support the rootstock coin type. So that wallet could use the correct derivation path [here](https://github.com/AlphaWallet/alpha-wallet-android/blob/10600581735c804376760277589fe9652953458d/app/src/main/java/com/alphawallet/app/service/KeyService.java#L402)

The purpose of this pull request is not only to list rootstock network but to support the coin type as there are many other wallets using this sdk. 

I request to reconsider this pull request as it will open doors for rootstock network integration with multiple wallets using this sdk.  🙏

File: registry.json [hunk 1]
```diff
@@ -2599,6 +2599,36 @@
       "documentation": "https://eth.wiki/json-rpc/API"
     }
   },
+  {
+    "id": "rootstock",
+    "name": "Rootstock",
+    "coinId": 137,
```
Comment (Milerius) [inline registry.json:2605]: please use 10000137 as id

**Review by Milerius — CHANGES_REQUESTED** [review]

File: registry.json (hunk 1 above)
Comment (ahsan-javaiid) [reply registry.json:2605]: CoinId is available for RSK(Roostock) and not used by any other chain. 
Reference: 
https://github.com/satoshilabs/slips/blob/master/slip-0044.md

```

Coin type | Path                | Symbol | Coin
-----------------------------------------
137           | 0x80000089 | RBTC.    | RSK
```

 Feel free to share if you have different thought's to use `10000137 ` as id.

**Comment by ahsan-javaiid** [comment]
> It's seems that tests are missing in CoinAddressDerivationTests kotlin/swift can you please fix @ahsan-javaiid

Added the tests. Thanks for pointing it out. 🙏

**Review by Milerius — APPROVED** [review]

**Pull request merged**